internal/controller/container_reconciler.go

```diff
@@ -250,8 +412,85 @@ func (r *ContainerImageReconciler) imageName(obj ContainerizedObject) string {
 	switch name := r.CloudContext.Name; name {
 	case cloud.GCP:
 		// Assuming this is Google Artifact Registry named "substratus".
-		return fmt.Sprintf("%s-docker.pkg.dev/%s/substratus/%s-%s-%s", r.CloudContext.GCP.Region(), r.CloudContext.GCP.ProjectID, strings.ToLower(r.Kind), obj.GetNamespace(), obj.GetName())
+		return fmt.Sprintf("%s-docker.pkg.dev/%s/substratus/%s-%s-%s",
+			r.CloudContext.GCP.Region(),
+			r.CloudContext.GCP.ProjectID,
+			strings.ToLower(r.Kind),
+			obj.GetNamespace(),
+			obj.GetName(),
+		)
 	default:
 		panic("unsupported cloud: " + name)
 	}
 }
+
+func (r *ContainerImageReconciler) signedUrlObjectName(obj ContainerizedObject) string {
+	return fmt.Sprintf("uploads/%s/%s/%s-%s-%s.tar.gz",
+		r.CloudContext.GCP.Region(),
+		r.CloudContext.GCP.ProjectID,
+		strings.ToLower(r.Kind),
+		obj.GetNamespace(),
+		obj.GetName(),
```

Comment:
Unsure what exactly we want for the .tar.gz upload location. This was specific enough for the container image so likely also the upload.

Reply:
sgtm